### DISCUSSION TITLE
Using this PS2 USB keyboard adaptor on other 8 bit systems

### DISCUSSION THREAD

**Discussion by banman11** [Hartland/C64-Keyboard#3] [Q&A]
Hypotheticaly speaking with the keymapping.h file modified appropriately, say a TRS 80 COCO 2 keyboard could be emulated?

A standard unmodified TRS80 COCO2 has only uppercase letters and a few shifted keys...

**Comment by Hartland** [comment]
In theory, I do not see why it would not work. There maybe some hardware changes that need to be made,

After I get setup for these type of projects again, I will add looking into this.

My main issue at this point is lack of these systems to design and test. I only have C64's at this time.
If we could find some hobbyists with these systems that have enough experience and skill to make protoboards, I would be more confidant about tackling it.

**Comment by banman11** [comment]
Hi Hartland,

That is very interesting. 
It just happens that I have here 2 X COCO 2 machines that function. One with a working keyboard and one without. I also have a Tandy MC-10 that I hope to have up and running soon.

I have some basic hobbyist skills (not very good) and I have a working PS2 style C64 keyboard adapter that works okay on a C64 main board.
I would be happy to explore this option of getting it to work on the COCO2 machines. Possibility the Tandy MC-10 if I can get working again.

I would of course need help from you and your intimate understanding of your amazing project.

By the way I did have success at running your project on a C16. It just needed some simple wiring redirections on the keyboard strip. I believe the C16/ plus4 uses the same matrix as the C64/VIC 20. It's just wired to different positions on the keyboard connector.

It doesn't need to be explored immediately. I too have real world work expectations that need addressing. 
A leisurely pace is good with me.

**Reply by Hartland** [reply]
I just acquired a COCO2 and it is the pipeline for development .

One of the reasons I separated the control code for the MT88XX was to make it easier for myself and others to develop keyboard adapters for other systems that use matrix type keyboards.

**Comment by levo-man** [comment]
I have several systems that this could be used for , Including Atari 800xl , dragon32 , several versions of spectrum and ZX81 .

Please have a look at my website for a build guide for the v1 ps/2 version .

[levosretrocomputerprojects.co.uk](https://levosretrocomputerprojects.co.uk/)

**Reply by banman11** [reply]
Nice website Levoman. 😁

**Reply by levo-man** [reply]
Thanks mate , are you the same Banman who helped with testing ?

**Comment by banman11** [comment]
Hi Levoman, 

Yes, that's right. I hope you are well 
It's a shame Melon64 got brought down by hackers. So much stuff was lost. A real shame.
 A good thing you have stuff preserved on your site, well done.
Actually, I got my hands on a C128. I am keen on getting your niffty little keyboard adapter working on that.
I also have a Tandy COCO2 that I want to modify the keymapping.h file (I think I have it right) and get your adapter working with it. I very kind gentleman by the name of Sensorium has been talking me thru the code.
https://github.com/SensoriumEmbedded

**Comment by levo-man** [comment]
Nice to hear from you , I hope you are well .

The coding is fairly easy , when you understand the language properly 
and how it affects the inputs/outputs on the controllers .

Let me know if you need any help with anything , or if there is anything 
I can do for you !



------ Original Message ------
From "banman11" ***@***.***>
To "Hartland/C64-Keyboard" ***@***.***>
Cc "levo-man" ***@***.***>; "Comment" 
***@***.***>
Date 07/08/2024 04:58:26
Subject Re: [Hartland/C64-Keyboard] Using this PS2 USB keyboard adaptor 
on other 8 bit systems (Discussion #3)

>
>Hi Levoman,
>
>Yes, that's right. I hope you are well
>It's a shame Melon64 got brought down by hackers. So much stuff was 
>lost. A real shame.
>A good thing you have stuff preserved on your site, well done.
>Actually, I got my hands on a C128. I am keen on getting your niffty 
>little keyboard adapter working on that.
>I also have a Tandy COCO2 that I want to modify the keyboard.h file (I 
>think I have it right) and get your adapter working with it. I very 
>kind gentleman by the name of Sensorium has been talking me thru the 
>code.
>https://github.com/SensoriumEmbedded
>
>—
>Reply to this email directly, view it on GitHub 
><https://github.com/Hartland/C64-Keyboard/discussions/3#discussioncomment-10259908>, 
>or unsubscribe 
><https://github.com/notifications/unsubscribe-auth/ADVWGAMUHGE5KR34GKK5N43ZQGLOFAVCNFSM6AAAAABE22DNECVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMRVHE4TAOA>.
>You are receiving this because you commented.Message ID: 
>***@***.***>
>

-- 
This email has been checked for viruses by Avast antivirus software.
www.avast.com

**Reply by banman11** [reply]
Absolutely Levoman!

**Comment by bugjacobs** [comment]
Would like a version for C128 too :-) Wedge model.

**Reply by banman11** [reply]
Hi bugjacobs,

I think one of these with Levoman's keyboard adapter should be fine. I know there's not a lot of wiggle room in those wedge C128's. I think Levoman's got one to fit in C64c wedge. So it might be possible.

https://www.pcbway.com/project/shareproject/COMMODORE_64_KEYBOARD_to_COMMODORE_128D_ADAPTER_68322010.html

**Reply by levo-man** [reply]
Sadly I do not have a c128 to test this with , otherwise I would have done a re-work to make it fit the 128 by now .

**Reply by Hartland** [reply]
I found a project based on a forked version of my project at:
https://github.com/ytmytm/C128-keyboard

I have only quickly scanned it, but it looks promising. It uses version 1 code, maybe I will need to make the author aware of version 2.
